File: level_1/하샤드 수.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

bool solution(int x)
{
    int imsy = x;
    int num = 0;
    while (imsy)
    {
        num += imsy % 10;
        imsy /= 10;
    }
    if (x % num == 0)
        return true;
    return false;
}

File: level_1/하샤드 수.java
class Solution {
    public boolean solution(int x) {
        int answer = 0;
        int n = x;
        while (n > 0) {
            answer += n % 10;
            n /= 10;
        }
        return x % answer == 0 ? true : false;
    }
}
